vmngclient/api/templates.py

```diff
@@ -0,0 +1,240 @@
+
+    def wait_complete(self, operation_id: str, timeout_seconds: int = 300, sleep_seconds: int = 5) -> bool:
+        """Wait to complete action.
+
+        Args:
+            operation_id (str): Id of the action waiting for completion.
+            timeout_seconds (int, optional): Time for completion. Defaults to 300.
+            sleep_seconds (int, optional): Sleep time between repetitions. Defaults to 5.
+
+        Returns:
+            bool: True if acction status is successful, otherwise - False.
+        """
+
+        def check_status(action_data):
+            if action_data:
+                list_action = [action == OperationStatus.SUCCESS for action in action_data]
+                return not all(list_action)
+            else:
+                return True
+
+        @retry(
+            wait=wait_fixed(sleep_seconds),
+            stop=stop_after_attempt(int(timeout_seconds / sleep_seconds)),
+            retry=retry_if_result(check_status),
+        )
+        def wait_for_status():
+            return self.get_operation_status(operation_id)
+
+        return True if wait_for_status() else False
+
+    def attach(self, template_name: str, device: DeviceInfo) -> bool:
+        """Attach template to device.
+
+        Args:
+            template_name (str): Template name to attached.
+            device (DeviceInfo): Device to attach template.
+
+        Returns:
+            bool: True if attached template is successful, otherwise - False.
+        """
+        templateId = self.get_id(template_name)
+        payload = {
+            "deviceTemplateList": [
+                {
+                    "templateId": templateId,
+                    "device": [
+                        {
+                            "csv-status": "complete",
+                            "csv-deviceId": device.uuid,
+                            "csv-deviceIP": device.id,
+                            "csv-host-name": device.hostname,
+                            "csv-templateId": templateId,
+                        }
+                    ],
+                }
+            ]
+        }
+        endpoint = "/dataservice/template/device/config/attachcli"
+        response = cast(dict, self.session.post_json(url=endpoint, data=payload))
+        return self.wait_complete(response['id'])
+
+    def device_to_cli(self, device: DeviceInfo) -> bool:
+        """Device mode to Cli.
+
+        Args:
+            device (DeviceInfo): Device to chcange mode.
+
+        Returns:
+            bool: True if change mode to cli is successful, otherwise - False.
+        """
+        payload = {"deviceType": device.personality, "devices": [{"deviceId": device.uuid, "deviceIP": device.id}]}
+        endpoint = "/dataservice/template/config/device/mode/cli"
+        response = cast(dict, self.session.post_json(url=endpoint, data=payload))
+        return self.wait_complete(response['id'])
+
+    def get_operation_information(self, id: str) -> List[Dict[str, str]]:
```

Comment:
maybe make it private method, so user don't get raw data dict without typing information?
Why not make dataclass for operation information?